### PR TITLE
Update Shield Options

### PR DESCRIPTION
**Pull request by sidschingis**
Use Guard Clauses to return early.
This reduces the maximum nesting level to create a more linear flow

https://refactoring.guru/replace-nested-conditional-with-guard-clauses

PS:
I don't have experience with Rust itself yet, so let me know if I made any syntax errors

### CODE REVIEW

**Comment by sidschingis** [comment]
- Updated infinite shield mode
    - Now takes normal shield damage
    - Will start deteriorating after getting hit
- Added OOS Offset option
    - Allow delaying OOS options until a certain number of hits have connected on shield
- Added forward and backwards roll mash options

File: src/hitbox_visualizer/mod.rs [hunk 1]
```diff
@@ -233,24 +237,9 @@ unsafe fn handle_attack(lua_state: u64) {
     let y2 = l2c_agent.pop_lua_stack(14); // float or void
     let z2 = l2c_agent.pop_lua_stack(15); // float or void
 
-    // hacky way of forcing no shield damage on all hitboxes
```
Comment (jugeeya) [inline src/hitbox_visualizer/mod.rs:236]: Did you reimplement this anywhere? Or just remove it?

File: src/training/directional_influence.rs [hunk 2]
```diff
@@ -12,32 +12,41 @@ pub static NO_DI : f64 = -69.0;
 
 #[skyline::hook(replace = smash::lua2cpp::L2CFighterCommon_FighterStatusDamage__correctDamageVectorCommon)]
 pub unsafe fn handle_correct_damage_vector_common(fighter: &mut L2CFighterCommon, arg1: L2CValue) -> L2CValue {
+    if !is_training_mode()
+    {
```
Comment (jugeeya) [inline src/training/directional_influence.rs:16]: nit: curly brace not on the same line

File: src/training/ledge.rs [hunk 3]
```diff
@@ -91,8 +107,18 @@ pub unsafe fn get_command_flag_cat(
...
-        defensive_option(module_accessor, category, flag);
+    if !is_training_mode(){
+        return;
+    }
+    
+    if !is_operation_cpu(module_accessor){
+        return;
+    }
+
+    if MENU.ledge_state == LedgeOption::None{
```
Comment (jugeeya) [inline src/training/ledge.rs:118]: nit: space before curly brace (for everything in this function)

File: src/training/save_states.rs [hunk 4]
```diff
@@ -30,163 +30,167 @@ static mut SAVE_STATE_SITUATION_KIND_CPU: i32 = 0 as i32;
...
-            save_state = &mut SAVE_STATE_CPU_STATE;
-        } else {
-            save_state_x = &mut SAVE_STATE_X_PLAYER;
-            save_state_y = &mut SAVE_STATE_Y_PLAYER;
-            save_state_percent = &mut SAVE_STATE_PERCENT_PLAYER;
-            save_state_lr = &mut SAVE_STATE_LR_PLAYER;
-            save_state_situation_kind = &mut SAVE_STATE_SITUATION_KIND_PLAYER;
-            save_state = &mut SAVE_STATE_PLAYER_STATE;
+
+    if !is_training_mode(){
```
Comment (jugeeya) [inline src/training/save_states.rs:34]: nit: space before curly brace

File: src/training/save_states.rs [hunk 5]
```diff
@@ -30,163 +30,167 @@ static mut SAVE_STATE_SITUATION_KIND_CPU: i32 = 0 as i32;
...
+        );
+
+        if *save_state_situation_kind == SITUATION_KIND_GROUND
+            && status != FIGHTER_STATUS_KIND_WAIT
+        {
+            StatusModule::change_status_request(
                 module_accessor,
-                -1.0 * DamageModule::damage(module_accessor, 0),
-                0,
+                *FIGHTER_STATUS_KIND_CLIFF_WAIT,
```
Comment (jugeeya) [inline src/training/save_states.rs:129]: This is wrong, should be `FIGHTER_STATUS_KIND_WAIT`. This is for if you're resetting either player to the idle state.

File: src/training/shield.rs [hunk 6]
```diff
@@ -8,154 +8,308 @@ use smash::app::sv_system;
 use smash::lib::L2CValue;
 use smash::lua2cpp::L2CFighterCommon;
 
+// Toggle for infinite shield decay
+static mut SHIELD_FLAG: bool = false;
+
+
+// How many hits to hold shield until picking an OOS option
+static mut MULTI_HIT_OFFSET : u8 = MENU.oos_offset;
+//
```
Comment (jugeeya) [inline src/training/shield.rs:17]: nit: is there meant to be a comment here?

File: src/training/shield.rs [hunk 7]
```diff
@@ -8,154 +8,308 @@ use smash::app::sv_system;
...
+}
+
+pub unsafe fn allow_oos()->bool {
+    // Delay OOS
+    MULTI_HIT_OFFSET == 0
+}
+
+unsafe fn handle_escape_mash(module_accessor: &mut app::BattleObjectModuleAccessor,fighter: &mut L2CFighterCommon)
+{
+    if MENU.mash_state == Mash::Spotdodge {
```
Comment (jugeeya) [inline src/training/shield.rs:204]: nit: `match` case?

**Pull request closed without merging**